m4/curl-functions.m4

```diff
@@ -2226,6 +2226,290 @@ AC_DEFUN([CURL_CHECK_FUNC_GETADDRINFO], [
+dnl defining HAVE_GETADDRINFO_A_THREADSAFE when true.
+
+AC_DEFUN([CURL_CHECK_FUNC_GETADDRINFO_A], [
+  AC_REQUIRE([CURL_INCLUDES_WS2TCPIP])dnl
+  AC_REQUIRE([CURL_INCLUDES_STDLIB])dnl
+  AC_REQUIRE([CURL_INCLUDES_STRING])dnl
+  AC_REQUIRE([CURL_INCLUDES_SYS_SOCKET])dnl
+  AC_REQUIRE([CURL_INCLUDES_NETDB])dnl
+  AC_REQUIRE([CURL_CHECK_NATIVE_WINDOWS])dnl
+  #
+  tst_links_getaddrinfo_a="unknown"
+  tst_proto_getaddrinfo_a="unknown"
+  tst_compi_getaddrinfo_a="unknown"
+  tst_works_getaddrinfo_a="unknown"
+  tst_allow_getaddrinfo_a="unknown"
+  tst_tsafe_getaddrinfo_a="unknown"
+  #
+  AC_MSG_CHECKING([if getaddrinfo_a can be linked])
+  AC_LINK_IFELSE([
+    AC_LANG_PROGRAM([[
+      $curl_includes_ws2tcpip
+      $curl_includes_sys_socket
+      $curl_includes_netdb
+    ]],[[
+      if(0 != getaddrinfo_a(0, 0, 0, 0))
+        return 1;
+    ]])
+  ],[
+    AC_MSG_RESULT([yes])
+    tst_links_getaddrinfo_a="yes"
+  ],[
+    tmp_LIBS="$LIBS"
+    LIBS="-lanl"
+    AC_LINK_IFELSE([
+      AC_LANG_PROGRAM([[
+        $curl_includes_ws2tcpip
+        $curl_includes_sys_socket
+        $curl_includes_netdb
+      ]],[[
+        if(0 != getaddrinfo_a(0, 0, 0, 0))
+          return 1;
+      ]])
+    ],[
+      AC_MSG_RESULT([yes])
+      tst_links_getaddrinfo_a="yes"
+    ],[
+      AC_MSG_RESULT([no])
+      tst_links_getaddrinfo_a="no"
+      LIBS=
+    ])
+    LIBS="$LIBS $tmp_LIBS"
+  ])
+  #
+  if test "$tst_links_getaddrinfo_a" = "yes"; then
+    AC_MSG_CHECKING([if getaddrinfo_a is prototyped])
+    AC_EGREP_CPP([getaddrinfo_a],[
+      $curl_includes_ws2tcpip
+      $curl_includes_sys_socket
+      $curl_includes_netdb
+    ],[
+      AC_MSG_RESULT([yes])
+      tst_proto_getaddrinfo_a="yes"
+    ],[
+      AC_MSG_RESULT([no])
+      tst_proto_getaddrinfo_a="no"
+    ])
+  fi
+  #
+  if test "$tst_proto_getaddrinfo_a" = "yes"; then
+    AC_MSG_CHECKING([if getaddrinfo_a is compilable])
+    AC_COMPILE_IFELSE([
+      AC_LANG_PROGRAM([[
+        $curl_includes_ws2tcpip
+        $curl_includes_sys_socket
+        $curl_includes_netdb
+      ]],[[
+        if(0 != getaddrinfo_a(0, 0, 0, 0))
+          return 1;
+      ]])
+    ],[
+      AC_MSG_RESULT([yes])
+      tst_compi_getaddrinfo_a="yes"
+    ],[
+      AC_MSG_RESULT([no])
+      tst_compi_getaddrinfo_a="no"
+    ])
+  fi
+  #
+  dnl only do runtime verification when not cross-compiling
+  if test "x$cross_compiling" != "xyes" &&
+    test "$tst_compi_getaddrinfo_a" = "yes"; then
+    AC_MSG_CHECKING([if getaddrinfo_a seems to work])
+    CURL_RUN_IFELSE([
+      AC_LANG_PROGRAM([[
+        $curl_includes_ws2tcpip
+        $curl_includes_stdlib
+        $curl_includes_string
+        $curl_includes_sys_socket
+        $curl_includes_netdb
+      ]],[[
+        struct addrinfo hints;
+        struct gaicb cb, *arr[1];
+        int error;
+
+        #ifdef HAVE_WINSOCK2_H
+        WSADATA wsa;
+        if (WSAStartup(MAKEWORD(2,2), &wsa))
```

Comment:
Winsock2 in conjunction with `getaddr_info_a()`. I don't get it. I thought `getaddr_info_a()` is a Posix / Linux thing.
The closest thing on Windows could probably be `DnsQueryEx()` with a callback.

Reply:
Oh, sorry. I forgot to re-review that part. This is still just a copy of the getaddrinfo check. My bad!

Reply:
Cleaned up.